migrator/migrations/n_05_to_n_06_postgres_active_components/migration.go

```diff
@@ -119,6 +136,28 @@ func storeWalk(ctx context.Context, s legacy.Store, fn func(obj *storage.ActiveC
 	return nil
 }
 
+func prune(postgresDB *pgxpool.Pool) error {
+	ctx := sac.WithAllAccess(context.Background())
+	deleteStmt := `DELETE FROM active_components child WHERE NOT EXISTS 
+		(SELECT * from deployments parent WHERE child.deploymentid = parent.id)`
+	_, err := postgresDB.Exec(ctx, deleteStmt)
+	if err != nil {
+		log.WriteToStderrf("failed to clean up orphaned data for %s", schema.Table)
+		return err
+	}
+
+	// Now get the child
+	deleteStmt = `DELETE FROM active_components_active_contexts_slices child WHERE NOT EXISTS 
```

Comment:
This is not necessary as long as triggers are on due to delete cascade

Reply:
@connorgorman Since I turned them off in Gorm before applying the schema for the parent table.  Gorm then applied the necessary table and children with foreign keys off.  So they are off when I move the data in.  Tradeoffs.  If we do this via Gorm it will turn the keys off for everything it creates in `pkgSchema.ApplySchemaForTable(context.Background(), gormDB, schema.Table)`. Then when we turn them back on, it will reapply them and thus ensure that all the data is properly constrained.  

We could be more finely controlled using sql with an alter on just active_components which would not impact its child.    Simple enough change to make if that is the preference.  Instead of disabling triggers I would drop the constraint which is what Gorm is doing behind the scenes.  That way we find out at that time if we have reasonable data integrity.

Reply:
@dashrews78 The thought then is that we should just do this after we have re-enabled triggers?

Reply:
if we do it with SQL by disabling triggers, then yes.  We could drop the constraint instead.  That would work in the same order as this version.  That might be simpler and more obvious.  It also wouldn't have any side effects with other things like unique constraints.  I'm thinking if we want to do it more controlled, that would probably be a better option